saiph/projection.py

```diff
@@ -51,6 +49,10 @@ def fit(
             )
 
     _nf = nf if nf else min(pd.get_dummies(df, prefix_sep=DUMMIES_PREFIX_SEP).shape)
+    # If seed is None or int, we fit a Generator, else we use the one provided.
+    random_gen = (
+        seed if isinstance(seed, np.random.Generator) else np.random.default_rng(seed)
```

Comment:
(this line does not feel right, hence my comment above)

Reply:
Can you explain more?